src/test/enginesynctest.cpp

```diff
@@ -1538,7 +1642,7 @@ TEST_F(EngineSyncTest, QuantizeImpliesSyncPhase) {
 
     // Deck 1 continues with 100 bpm
     EXPECT_DOUBLE_EQ(100, ControlObject::get(ConfigKey(m_sGroup1, "bpm")));
-    EXPECT_DOUBLE_EQ(0.044504913076341648, ControlObject::get(ConfigKey(m_sGroup1, "beat_distance")));
+    EXPECT_DOUBLE_EQ(0.050309901738473169, ControlObject::get(ConfigKey(m_sGroup1, "beat_distance")));
```

Comment:
This is actually a test failure. 
0.025154950869236584 * 2 = 0.050309901738473169
so the test continues with 130 BPM which is wrong. 

0.044504913076341648 - 0.025154950869236584 = 0,019349962
0,019349962 / 0.025154950869236584 * 130 = 100; 

So i am afraid you need to revert to 0.044504913076341648 and fix the code.

Reply:
there is a one buffer delay between the detection of a sync mode change and when it's activated. EngineBuffer::requestSyncMode sets syncmodequeued.  Let me see if this is an issue in master

Reply:
ah, no, that's not it. still looking

Reply:
ah, got it.  sync requests have to be processed before we grab the rate.